### ISSUE TITLE
I'm not able to use this package with stylelint

### ISSUE THREAD

**Issue by alokVishu** [stylelint-stylistic/stylelint-stylistic#1]
Package version:
```
devDependencies:
+ stylelint-codeguide 0.2.0

 WARN  Issues with peer dependencies found
.
├─┬ stylelint-config-idiomatic-order 9.0.0
│ └─┬ stylelint-order 5.0.0
│   └── ✕ unmet peer stylelint@^14.0.0: found 15.10.0
└─┬ vite-plugin-vue-devtools 0.4.12
  └─┬ @webfansplz/vuedoc-parser 0.0.3
    └── ✕ unmet peer @types/node@^18.6.3: found 20.3.3
```

Here is my stylelintrc.json file
```
{
  "extends": [
    "stylelint-config-standard-scss",
    "stylelint-config-idiomatic-order"
  ],
  "plugins": [
    "stylelint-use-logical-spec",
    "stylelint-codeguide"
  ],
  "overrides": [
    {
      "files": [
        "**/*.scss"
      ],
      "customSyntax": "postcss-scss"
    },
    {
      "files": [
        "**/*.vue"
      ],
      "customSyntax": "postcss-html"
    }
  ],
  "rules": {
    "codeguide/max-line-length": [
      120,
      {
        "ignore": "comments"
      }
    ],
    "liberty/use-logical-spec": true,
    "selector-class-pattern": null,
    "color-function-notation": null
  }
}
```

And here is my stylelint out:
```
[Info  - 13:42:37] [language-server] Registering module | module: "auto-fix"
[Info  - 13:42:38] [language-server] Module registered | module: "auto-fix"
[Info  - 13:42:38] [language-server] Registering module | module: "code-action"
[Info  - 13:42:38] [language-server] Module registered | module: "code-action"
[Info  - 13:42:38] [language-server] Registering module | module: "completion"
[Info  - 13:42:38] [language-server] Module registered | module: "completion"
[Info  - 13:42:38] [language-server] Registering module | module: "formatter"
[Info  - 13:42:38] [language-server] Module registered | module: "formatter"
[Info  - 13:42:38] [language-server] Registering module | module: "old-stylelint-warning"
[Info  - 13:42:38] [language-server] Module registered | module: "old-stylelint-warning"
[Info  - 13:42:38] [language-server] Registering module | module: "validator"
[Info  - 13:42:38] [language-server] Module registered | module: "validator"
[Info  - 13:42:38] [language-server] Starting language server
[Info  - 13:42:38] [language-server] Registering handlers
[Info  - 13:42:38] [language-server] Handlers registered
[Info  - 13:42:38] [language-server] Language server started
/Users/aloksoni/Projects/clevision/master/vue/typescript-version/full-version/src/@core/scss/base/_vertical-nav.scss: you should use the "customSyntax" option when linting something other than CSS
```

No indentation and nothing works.

**Comment by firefoxic** [comment]
Hi! I could change the version to `^15.0.0`. But that doesn't seem to solve your problem, because `stylelint-order` requires `stylelint@^14.0.0`.

And I deliberately removed `stylelint@^14.16.1` from `peerDependencies` with a separate release that doesn't change anything significant except that dependency. This is so that you can use `0.1.5` to gradually move from `stylelint@14` to `stylelint@15`. You can periodically check if all dependencies are working by updating this plugin to version `0.2.0` and rolling back to `0.1.5` if there are still conflicts.

By the way, major version `1.0.0` will be released after the `stylelint@16` release, which will remove the obsolete rules altogether.

In summary, use `stylelint-codeguide@~0.1.5` and let me know if your tools work with it or not.

P.S. Perhaps I should have written a migration guide 🤔

**Comment by firefoxic** [comment]
> I have also reproduced the same issue in a fresh blank project. Please find the attached zip file.

In the archive, `stylelint-codeguide` is not involved in the conflict.

Even if you remove this plugin completely, you still have dependency conflicts. `stylelint-config-idiomatic-order` depends on `stylelint@^14`, and `stylelint-config-standard-scss` depends on `stylelint@^15+`.

You should then roll back `stylelint-config-standard-scss` to the version that depends on `stylelint@14`. Then roll back `stylelint` itself to `14.16.1`. And you can use `stylelint-codeguide@~0.1.5` in that set.

If you really need `stylelint-config-idiomatic-order`, you will have to wait for it to update and require `stylelint@^15`.

**Comment by firefoxic** [comment]
Your archives are missing a file called `.stylelintrc'.

Try this [Archive.zip](https://github.com/firefoxic/stylelint-codeguide/files/11958281/Archive.zip). I updated all dependencies and added config. Stylelint is correctly indicating indentation errors.
![image](https://github.com/firefoxic/stylelint-codeguide/assets/3382798/c4ef4bcc-3335-42d9-9ae9-6515dba0f56e)

**Comment by jd-solanki** [comment]
@firefoxic can you provide config list with recommended settings in your package along with rules so others can extend instead of writing long list of rules?

**Comment by firefoxic** [comment]
> @firefoxic can you provide config list with recommended settings in your package along with rules so others can extend instead of writing long list of rules?

I have my own [config](https://www.npmjs.com/package/stylelint-config-firefoxic) that I use.
